app/views/play/menu/InventoryModal.coffee

```diff
@@ -656,8 +655,8 @@ module.exports = class InventoryModal extends ModalView
       @setUpDraggableEventsForAvailableEquipment()
       @itemDetailsView.setItem(item)
       @onScrollUnequipped true
-
-      Backbone.Mediator.publish 'store:item-purchased', item: item, itemSlug: item.get('slug')
+      if not me.isStudent()
```

Comment:
This check is could be more specific, also checking `me.showHeroAndInventoryModalsToStudents`.